### PR TITLE
Storage Copy code

### PR DESCRIPTION
**Pull request by anjali411**
Investigating if the cross dtype storage code is dead

### CODE REVIEW

**Comment by anjali411** [comment]
cc. @ezyang

File: torch/csrc/generic/Storage.cpp [hunk 1]
```diff
@@ -308,25 +308,35 @@ void THPStorage_(initCopyMethods)()
 {
   auto& h = THWStorage_(copy_functions);
   // copy from CPU types
-  THPInsertStorageCopyFunction<THPStorage, THPStorage>(&THPByteStorageType, h, &THWStorage_(copyByte));
```
Comment (ezyang) [inline torch/csrc/generic/Storage.cpp:311]: D'oh, that's why I couldn't find this code; it was `THWStorage_` lol

**Comment by ezyang** [comment]
Per the test in https://github.com/pytorch/pytorch/pull/35851/files#diff-9996665f82f52030836eb8657057cfadL3973 it would seem that it is used solely by the Python API on storage. So, strictly speaking, removing it would be a BC-breaking change.

What is interesting to know is if Tensor has a completely disjoint set of code for doing cross-dtype conversions. We only need one; if you can dtype convert Tensor, then you can dtype convert Storage by making a Tensor around the storage, dtype converting the Tensor, and then returning the storage of the resulting tensor. If that's true, then we can still remove this code while maintaining BC.

Also cc @VitalyFedyunin who was also looking into replacing all of the Storage Python functionality with corresponding Tensor functionality, a long time ago.

**Comment by VitalyFedyunin** [comment]
> What is interesting to know is if Tensor has a completely disjoint set of code for doing cross-dtype conversions. We only need one; if you can dtype convert Tensor, then you can dtype convert Storage by making a Tensor around the storage, dtype converting the Tensor, and then returning the storage of the resulting tensor. If that's true, then we can still remove this code while maintaining BC.

I like this idea, as Tensor's type conversion currently look like: create new tensor (and new storage) with requested type, copy (casting) data from source tensor to the destination.

Best case scenario is to move storage casting operations from cpp code and do this tensor case+return new storage on python lavel of `Storage` class.

**Pull request closed without merging**